shared/utils/il2cpp-utils.hpp

```diff
@@ -146,104 +146,62 @@ namespace il2cpp_utils {
-        if (exp) {
-            log(ERROR, "il2cpp_utils: New: Failed with exception: %s", ExceptionToString(exp).c_str());
+        auto klass = il2cpp_functions::object_get_class(instance);
+        if (!klass) {
+            log(ERROR, "il2cpp_utils: %s: Could not find object's class!", whosAsking.data());
             return nullptr;
         }
-        return reinterpret_cast<TObj*>(obj);
+        return klass;
     }
 
-    template<typename TObj = Il2CppObject, typename... TArgs>
-    // Creates a New object of the given class and parameters and casts it to TObj*
-    // DOES NOT PERFORM TYPE-SAFE CHECKING!
-    TObj* NewUnsafe(Il2CppClass* klass, TArgs* ...args) {
-        il2cpp_functions::Init();
+    // Returns if a given MethodInfo's parameters match the Il2CppType array provided as type_arr
+    bool ParameterMatch(const MethodInfo* method, const Il2CppType* const* type_arr, decltype(MethodInfo::parameters_count) count);
 
-        void* invokeParams[] = {reinterpret_cast<void*>(args)...};
-        // object_new call
-        auto obj = il2cpp_functions::object_new(klass);
-        // runtime_invoke constructor with right number of args, return null if constructor errors
+    template<typename... TArgs>
+    // Returns if a given MethodInfo's parameters match the Il2CppTypes provided as args
+    bool ParameterMatch(const MethodInfo* method, TArgs* ...args) {
         constexpr auto count = sizeof...(TArgs);
-        log(DEBUG, "Attempting to find .ctor with paramCount: %lu for class name: %s", count, il2cpp_functions::class_get_name(klass));
-        const MethodInfo* ctor = il2cpp_functions::class_get_method_from_name(klass, ".ctor", count);
+        Il2CppType* argarr[] = {reinterpret_cast<Il2CppType*>(args)...};
+        return ParameterMatch(method, argarr, count);
+    }
 
-        if (!ctor) {
-            log(ERROR, "il2cpp_utils: New: Could not find constructor for provided class!");
-            return nullptr;
-        }
-        Il2CppException* exp;
-        il2cpp_functions::runtime_invoke(ctor, obj, invokeParams, &exp);
-        if (exp) {
-            log(ERROR, "il2cpp_utils: New: Failed with exception: %s", ExceptionToString(exp).c_str());
-            return nullptr;
-        }
-        return reinterpret_cast<TObj*>(obj);
+    // Returns if a given MethodInfo's parameters match the Il2CppTypes provided as a vector
+    inline bool ParameterMatch(const MethodInfo* method, std::vector<const Il2CppType*> seq) {
+        return ParameterMatch(method, seq.data(), seq.size());
     }
 
     // Returns the MethodInfo for the method on the given class with the given name and number of arguments
-    // TODO: HASH MAP KNOWN FUNCTIONS
     // Created by zoller27osu
     const MethodInfo* FindMethod(Il2CppClass* klass, std::string_view methodName, int argsCount);
+    inline const MethodInfo* FindMethod(Il2CppClass* klass, std::string_view methodName, size_t argsCount) {
```

Comment:
And this one uses size_t for argsCount... A bit of inconsistency, imo

Reply:
without this, attempting to call it with a size_t parameters count will resolve to the vector wrappers instead, because size_t is not the same as int.
I figured between int literals and sizeof...(TArgs), these 2 definitions should cover our bases.